website/docs/r/private_link_service.html.markdown

```diff
@@ -0,0 +1,150 @@
+}
+
+resource "azurerm_public_ip" "example" {
+  name                = "example-api"
+  sku                 = "Standard"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+  allocation_method   = "Static"
+}
+
+resource "azurerm_lb" "example" {
+  name                = "example-lb"
+  sku                 = "Standard"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+
+  frontend_ip_configuration {
+    name                 = azurerm_public_ip.example.name
+    public_ip_address_id = azurerm_public_ip.example.id
+  }
+}
+
+resource "azurerm_private_link_service" "example" {
+  name                = "myPrivateLinkService"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+
+  auto_approval_subscription_ids              = ["00000000-0000-0000-0000-000000000000"]
+  visibility_subscription_ids                 = ["00000000-0000-0000-0000-000000000000"]
+  load_balancer_frontend_ip_configuration_ids = [azurerm_lb.example.frontend_ip_configuration.0.id]
+
+  primary_nat_ip_configuration {
+    name                       = "primaryIpConfiguration"
+    private_ip_address         = "10.5.1.17"
+    private_ip_address_version = "IPv4"
+    subnet_id                  = azurerm_subnet.example.id
+  }
+
+  secondary_nat_ip_configuration {
+    name                       = "secondaryIpConfiguration"
+    private_ip_address         = "10.5.1.18"
+    private_ip_address_version = "IPv4"
+    subnet_id                  = azurerm_subnet.example.id
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the private link service. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) The name of the resource group in which the private link service resides. Changing this forces a new resource to be created.
+
+* `location` - (Required) Specifies the supported Azure location where the resource exists. Changing this forces a new resource to be created.
+
+* `auto_approval_subscription_ids` - (Optional) A list of subscription globally unique identifiers(GUID) that will be automatically be able to use this service.
+
+* `visibility_subscription_ids` - (Optional) A list of subscription globally unique identifiers(GUID) that will be able to see this service. If left undefined all Azure subscriptions will be able to see this service.
+
+* `primary_nat_ip_configuration` - (Required) A `primary_nat_ip_configuration` block as defined below.
+
+* `secondary_nat_ip_configuration` - (Optional) Supports upto seven additional `secondary_nat_ip_configuration` blocks as defined below.
+
+* `load_balancer_frontend_ip_configuration_ids` - (Required) A list of `Standard` Load Balancer resource ids to direct the service network traffic toward.
```

Comment:
Are the load balancers on the service or endpoint side?

Reply:
Service.

Reply:
👌